shared/AppInsightsCommon/src/Offline.ts

```diff
@@ -1,11 +1,27 @@
 import {
-    createUniqueNamespace, eventOff, eventOn, getDocument, getNavigator, getWindow, isNullOrUndefined, isUndefined, mergeEvtNamespace
+    IUnloadHook, createUniqueNamespace, eventOff, eventOn, getDocument, getNavigator, getWindow, isNullOrUndefined, isUndefined,
+    mergeEvtNamespace
 } from "@microsoft/applicationinsights-core-js";
 
+export type OfflineCallback = () => void;
+export const enum eOfflineValue {
+    Unknown = 0,
+    Online = 1,
+    Offline = 2
+}
+export interface IOfflineState {
+    readonly isOnline: boolean;
+    readonly rState: eOfflineValue; // runtime state
+    readonly uState: eOfflineValue;// user state
+}
+export declare function OfflineCallback(state?: IOfflineState): void;
```

Comment:
Please add comments above all of these to identify what they are / will be used for.

Likewise for all of the functions.